### PR TITLE
Exhaust scripts

### PR DESCRIPTION
**Pull request by manushak**

### CODE REVIEW

File: docs/developers/how-to-create-exhaust-script.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+sidebar-position: 6
+---
+
+# How to create an exhaust script
+
+The `If` framework provides 2 types of exhaust scripts: [`if-run`](../major-concepts/if.md) for outputting data in `yaml` format and [`if-csv`](../users/how-to-export-csv-file-with-if-csv.md) for outputting data in `csv` format. This guid will help you create your own exhaust script.
```
Comment (jmcook1186) [inline docs/developers/how-to-create-exhaust-script.md:7]:
```suggestion
The `If` framework provides 2 types of exhaust scripts: [`if-run`](../major-concepts/if.md) for outputting data in `yaml` format and [`if-csv`](../users/how-to-export-csv-file-with-if-csv.md) for outputting data in `csv` format. This guide will help you create your own exhaust script.
```

File: docs/developers/how-to-create-exhaust-script.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+sidebar-position: 6
+---
+
+# How to create an exhaust script
+
+The `If` framework provides 2 types of exhaust scripts: [`if-run`](../major-concepts/if.md) for outputting data in `yaml` format and [`if-csv`](../users/how-to-export-csv-file-with-if-csv.md) for outputting data in `csv` format. This guide will help you create your own exhaust script.
```
Comment (jmcook1186) [inline docs/developers/how-to-create-exhaust-script.md:7]:
```suggestion
The `If` framework outputs data in `yaml` format. Any other output formats require a separate script that takes the yaml output data and processes it. We provide [`if-csv`](../users/how-to-export-csv-file-with-if-csv.md) for outputting data in `csv` format bundled with IF.  For any other format, you need to write an exhaust script. 
This guide will help you create your own exhaust script.
```

File: docs/major-concepts/exhaust-script.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 10
+---
+
+# Exhaust scripts
+
+Exhaust script is the major concept of the `if`, which provides two types of exhaust scripts: `if-run` and `if-csv`.
```
Comment (jmcook1186) [inline docs/major-concepts/exhaust-script.md:7]: if-run isn't really an exhaust script -  it does observe/regroup/compute as well. I'd define an exhaust script as a script that takes a yaml file as input and outputs some other representation of the same data.

Comment (manushak) [reply docs/major-concepts/exhaust-script.md:7]: > if-run isn't really an exhaust script - it does observe/regroup/compute as well. I'd define an exhaust script as a script that takes a yaml file as input and outputs some other representation of the same data.

Thanks  @jmcook1186 for clarification!

Comment (jmcook1186) [reply docs/major-concepts/exhaust-script.md]: added some changes

**Pull request merged**